### PR TITLE
chore: Migrate /superset/tables/* to API v1

### PR DESCRIPTION
**Pull request by diegomedina248**
### SUMMARY

Continuing the effort on deprecating all `/superset/` REST API endpoints
Deprecates `/tables/<int:db_id>/<schema>/`
Deprecates `/tables/<int:db_id>/<schema>/<force_refresh>/`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/databases/schemas.py [hunk 1]
```diff
@@ -555,6 +563,12 @@ class SchemasResponseSchema(Schema):
     result = fields.List(fields.String(description="A database schema name"))
 
 
+class DatabaseTablesResponse(Schema):
+    extra = fields.Dict(required=False)
+    type = fields.String()
+    value = fields.String()
```
Comment (dpgaspar) [inline superset/databases/schemas.py:569]: would be nice to add a description on these fields to improve out documentation for the API

File: superset/databases/schemas.py [hunk 2]
```diff
@@ -555,6 +563,12 @@ class SchemasResponseSchema(Schema):
     result = fields.List(fields.String(description="A database schema name"))
 
 
+class DatabaseTablesResponse(Schema):
+    extra = fields.Dict(required=False)
```
Comment (dpgaspar) [inline superset/databases/schemas.py:567]: don't think that the require=False is necessary here

File: superset/databases/commands/tables.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    {
+                        "value": view.table,
+                        "type": "view",
+                    }
+                    for view in views
+                ],
+                key=lambda item: item["value"],
+            )
+
+            payload = {"tableLength": len(tables) + len(views), "options": options}
```
Comment (dpgaspar) [inline superset/databases/commands/tables.py:103]: If `tableLength` is a count of all records on the payload, we are using `count` on all other API endpoints

File: superset/databases/api.py [hunk 4]
```diff
@@ -493,6 +500,76 @@ def schemas(self, pk: int, **kwargs: Any) -> FlaskResponse:
...
+    @statsd_metrics
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **kwargs: f"{self.__class__.__name__}" f".tables",
+        log_to_statsd=False,
+    )
+    def tables(self, pk: int, **kwargs: Any) -> FlaskResponse:
+        """Get a list of tables for given database
+        ---
+        get:
+          description: Get a list of tables for given database
```
Comment (dpgaspar) [inline superset/databases/api.py:516]: let's start using `summary:` instead of `description:`

**Review by dpgaspar — COMMENTED** [review]
Looks good!

File: superset/databases/api.py [hunk 5]
```diff
@@ -493,6 +500,76 @@ def schemas(self, pk: int, **kwargs: Any) -> FlaskResponse:
...
+        try:
+            command = TablesDatabaseCommand(pk, schema_name, force)
+            payload = command.run()
+            return self.response(200, result=payload)
+        except DatabaseNotFoundError:
+            return self.response_404()
+        except SupersetException as ex:
+            return self.response(ex.status, message=ex.message)
+        except DatabaseTablesUnexpectedError as ex:
+            return self.response_500(ex.message)
```
[the remark this answers is not in the document]
Comment (diegomedina248) [reply superset/databases/api.py:569]: it was like that in the deprecated endpoint, though we can certainly use any other status code. Which one would be best here, to maintain consistency across the api?

File: tests/integration_tests/databases/api_tests.py [hunk 6]
```diff
@@ -1381,6 +1381,56 @@ def test_database_schemas_invalid_query(self):
...
+        """
+        self.login(username="admin")
+        database = db.session.query(Database).filter_by(database_name="examples").one()
+
+        schema_name = self.default_schema_backend_map[database.backend]
+        rv = self.client.get(
+            f"api/v1/database/{database.id}/tables/?q={prison.dumps({'schema_name': schema_name})}"
+        )
+
+        self.assertEqual(rv.status_code, 200)
```
[the remark this answers is not in the document]
Comment (diegomedina248) [reply tests/integration_tests/databases/api_tests.py:1396]: Yeah, the problem with the payload here is that it varies from engine to engine.
should we mock the response at this level, or what do you recommend here?

File: superset/databases/api.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dpgaspar) [reply superset/databases/api.py:569]: HTTP 422 or 400

File: tests/integration_tests/databases/api_tests.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dpgaspar) [reply tests/integration_tests/databases/api_tests.py:1396]: I see, how about only checking for a specific engine like postgresql?

File: superset-frontend/src/hooks/apiResources/tables.ts [hunk 7]
```diff
@@ -39,11 +40,16 @@ export interface Table {
 }
 
 type QueryData = {
-  json: { options: Table[]; tableLength: number };
+  json: {
+    result: {
+      options: Table[];
+      count: number;
+    };
+  };
```
Comment (dpgaspar) [inline superset-frontend/src/hooks/apiResources/tables.ts:48]: isn't this:
```
type QueryData = {
  json: {
    count: number;
    result: {
      options: Table[];
    };
  };
```

File: superset/databases/api.py [hunk 8]
```diff
@@ -550,6 +557,76 @@ def schemas(self, pk: int, **kwargs: Any) -> FlaskResponse:
...
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      result:
+                        description: >-
+                          A List of tables for given database
+                        type: object
+                        properties:
+                          count:
```
Comment (dpgaspar) [inline superset/databases/api.py:599]: bring the `count` out of the `result` object`, like:
``` json
{
    count: 10,
    result: [
          {
                 ....
          },
          {
                ....
          },
          ...
    ]
}
```

**Review by dpgaspar — COMMENTED** [review]
One last comment :)

**Review by dpgaspar — APPROVED** [review]
LGTM, after fixing conflicts

**Review by geido — APPROVED** [review]
Tests LGTM!

**Pull request merged**